app/src/main/java/com/kickstarter/viewmodels/ThanksShareHolderViewModel.kt

```diff
@@ -0,0 +1,122 @@
+package com.kickstarter.viewmodels
+
+import android.util.Pair
+import com.kickstarter.libs.ActivityViewModel
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.RefTag.Companion.thanksFacebookShare
+import com.kickstarter.libs.RefTag.Companion.thanksShare
+import com.kickstarter.libs.RefTag.Companion.thanksTwitterShare
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.UrlUtils.appendRefTag
+import com.kickstarter.models.Project
+import com.kickstarter.ui.viewholders.ThanksShareViewHolder
+import rx.Observable
+import rx.subjects.BehaviorSubject
+import rx.subjects.PublishSubject
+
+interface ThanksShareHolderViewModel {
+    interface Inputs {
+        /** Call to configure the view model with a project.  */
+        fun configureWith(project: Project)
+
+        /** Call when the share button is clicked.  */
+        fun shareClick()
+
+        /** Call when the share on Facebook button is clicked.  */
+        fun shareOnFacebookClick()
+
+        /** Call when the share on Twitter button is clicked.  */
+        fun shareOnTwitterClick()
+    }
+
+    interface Outputs {
+        /** Emits the backing's project name.  */
+        fun projectName(): Observable<String>
+
+        /** Emits the project name and url to share using Android's default share behavior.  */
+        fun startShare(): Observable<Pair<String, String>>
+
+        /** Emits the project and url to share using Facebook.  */
+        fun startShareOnFacebook(): Observable<Pair<Project, String>>
+
+        /** Emits the project name and url to share using Twitter.  */
+        fun startShareOnTwitter(): Observable<Pair<String, String>>
+    }
+
+    class ViewModel(environment: Environment) :
+        ActivityViewModel<ThanksShareViewHolder?>(environment), Inputs, Outputs {
```

Comment:
ActivityViewModel<ThanksShareViewHolder?> could you remove the optional part there?